### PR TITLE
style: fix mypy failure

### PR DESCRIPTION
**Pull request by Lee-W**
https://github.com/astronomer/astronomer-providers/pull/1241

We'll need to wait for the release of https://github.com/apache/airflow/issues/32568

### CODE REVIEW

**Review by pankajkoti — APPROVED** [review]

**Comment by Lee-W** [comment]
@pankajkoti Thanks for reminding me! Messed it up during cherry-pick 🤦‍♂️

File: astronomer/providers/microsoft/azure/hooks/wasb.py [hunk 1]
```diff
@@ -113,11 +112,11 @@ def _get_container_client(self, container_name: str) -> ContainerClient:
     async def get_blobs_list_async(
         self,
         container_name: str,
-        prefix: Optional[str] = None,
-        include: Optional[List[str]] = None,
+        prefix: str | None = None,
+        include: list[str] | None = None,
         delimiter: str = "/",
         **kwargs: Any,
-    ) -> List[BlobProperties]:
```
Comment (pankajastro) [inline astronomer/providers/microsoft/azure/hooks/wasb.py:120]: I think we should inline this with https://github.com/apache/airflow/pull/32545 changes

Comment (Lee-W) [reply astronomer/providers/microsoft/azure/hooks/wasb.py:120]: Sure! I've updated it. Could you please check again? Thanks!

**Comment by Lee-W** [comment]
as https://github.com/apache/airflow/issues/32568 is merged, I'll merge this one now

**Pull request merged**